apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/monetary/service/AssetServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+/*
+ *  Copyright © 2018-2020 Apollo Foundation
+ */
+
+package com.apollocurrency.aplwallet.apl.core.monetary.service;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import java.util.stream.Stream;
+
+import com.apollocurrency.aplwallet.apl.core.app.Transaction;
+import com.apollocurrency.aplwallet.apl.core.converter.IteratorToStreamConverter;
+import com.apollocurrency.aplwallet.apl.core.db.DbClause;
+import com.apollocurrency.aplwallet.apl.core.db.DbIterator;
+import com.apollocurrency.aplwallet.apl.core.db.DbKey;
+import com.apollocurrency.aplwallet.apl.core.db.service.BlockChainInfoService;
+import com.apollocurrency.aplwallet.apl.core.monetary.dao.AssetTable;
+import com.apollocurrency.aplwallet.apl.core.monetary.model.Asset;
+import com.apollocurrency.aplwallet.apl.core.transaction.messages.ColoredCoinsAssetIssuance;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class AssetServiceImpl implements AssetService {
+
+    private AssetTable assetTable;
+    private BlockChainInfoService blockChainInfoService;
+    private AssetDeleteService assetDeleteService;
+    private IteratorToStreamConverter<Asset> assetIteratorToStreamConverter =
+        new IteratorToStreamConverter<>();
```

Comment:
Consider using constructor injection here

Reply:
@Inject
    public AssetDeleteServiceImpl(...
injection is used

Reply:
I'm talking about defing new IteratorToStreamConverter<>(); in a constructor rather than in `    private IteratorToStreamConverter<Asset> assetIteratorToStreamConverter =
        new IteratorToStreamConverter<>();`

Reply:
I'm not sure it's possible because of:
**@Vetoed**
public class IteratorToStreamConverter<T>

Reply:
@Inject
    public AssetServiceImpl(AssetTable assetTable,
                            BlockChainInfoService blockChainInfoService,
                            AssetDeleteService assetDeleteService,
                            **IteratorToStreamConverter<Asset> assetIteratorToStreamConverter**

Reply:
I've put that into constructor for unit tests mostly

Reply:
it's possible, please, use BidOrderServiceImpl as an example
`
public class BidOrderServiceImpl implements OrderService<BidOrder, ColoredCoinsBidOrderPlacement> {
    static final String ORDER = " ORDER BY price DESC, creation_height ASC, transaction_height ASC, transaction_index ASC ";
    private final DatabaseManager databaseManager;
    private final BidOrderTable bidOrderTable;
    private final IteratorToStreamConverter<BidOrder> converter;
    private final Blockchain blockchain;

    /**
     * Constructor for unit tests
     *
     * @param databaseManager
     * @param bidOrderTable
     * @param blockchain
     * @param converter
     */
    public BidOrderServiceImpl(
        final DatabaseManager databaseManager,
        final BidOrderTable bidOrderTable,
        final Blockchain blockchain,
        final IteratorToStreamConverter<BidOrder> converter
    ) {
        this.databaseManager = databaseManager;
        this.bidOrderTable = bidOrderTable;
        this.blockchain = blockchain;
        this.converter = converter;
    }
`

Reply:
and 
    @Inject
    public BidOrderServiceImpl(
        final DatabaseManager databaseManager,
        final BidOrderTable bidOrderTable,
        final Blockchain blockchain
    ) {
        this.databaseManager = databaseManager;
        this.bidOrderTable = bidOrderTable;
        this.blockchain = blockchain;
        this.converter = new IteratorToStreamConverter<>();
    }

Reply:
Updated code to 'final'